drivers/rkt/driver.go

```diff
@@ -445,18 +445,26 @@ func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *cstru
 	// Convert underscores to dashes in taskConfig names for use in volume names #2358
 	sanitizedName := strings.Replace(cfg.Name, "_", "-", -1)
 
+	// Extract allocID from TaskConfig.ID written like: fmt.Sprintf("%s/%s/%d", tr.allocID, tr.taskName, tr.restartTracker.GetCount()),
+	var allocID string
+	if splitID := strings.SplitN(cfg.ID, "/", 2); len(splitID) < 2 {
+		return nil, nil, fmt.Errorf("TaskConfig.ID '%s' was not formatted as expected already started", cfg.ID)
+	} else {
+		allocID = splitID[0]
+	}
```

Comment:
I don't think we want to rely on the task ID being stable, so let's do the extra work of adding the AllocID to the TaskConfig struct:

* TaskConfig go struct
 https://github.com/hashicorp/nomad/blob/6b7953af21417c28c8f4d78e7a031f86be0d842d/plugins/drivers/driver.go#L99
* TaskConfig proto message https://github.com/hashicorp/nomad/blob/6b7953af21417c28c8f4d78e7a031f86be0d842d/plugins/drivers/proto/driver.proto#L298-L304
* Util funcs https://github.com/hashicorp/nomad/blob/6b7953af21417c28c8f4d78e7a031f86be0d842d/plugins/drivers/utils.go#L48-L85 

And setting it here:
https://github.com/hashicorp/nomad/blob/6b7953af21417c28c8f4d78e7a031f86be0d842d/client/allocrunner/taskrunner/task_runner.go#L650-L666

@nickethier - Does this seem reasonable to you? Since the alloc id is tucked into so many other parameters opaquely (TaskID, env vars, etc), I can't imagine why plumbing it through properly would be an issue.

Reply:
Will follow this approach, thanks @schmichael